### PR TITLE
[Apple Silicon] [MLX] Reject --enable-mixed-chunk, which segfaults the scheduler

### PR DESCRIPTION
**Pull request by jlee5814**
## Motivation

On the MLX backend, `--enable-mixed-chunk` with an explicit `--chunked-prefill-size` segfaults the scheduler when a prefill request arrives while another request is decoding. The subprocess dies with exit code -11. The trace lands in `torch.cat` on MPS during batch preparation, before the prefill batch line is logged.

Repro: `mlx-community/Qwen3-0.6B-4bit`, `--device mps`, `--chunked-prefill-size 512`, `--enable-mixed-chunk`, a long generation in flight, then an oversized prompt (larger than the chunk size) arriving while it is still decoding.

```
Fatal Python error: Segmentation fault
...
File "python/sglang/srt/managers/overlap_utils.py", line 100 in resolve_forward_inputs
File "python/sglang/srt/hardware_backend/mlx/scheduler_mixin.py", line 165 in _launch_fresh
...
in at::native::structured_cat_out_mps::impl(...)
in at::(anonymous namespace)::wrapper_MPS_cat(...)
Subprocess scheduler_0 crashed with exit code -11.
```

`resolve_forward_inputs` (`overlap_utils.py:90-100`) takes its staging branch whenever `prefill_input_ids_cpu` is set, which `prepare_for_extend` (`schedule_batch.py:2178`) does for every extend batch on every backend. On a mixed batch it reads `future_map.output_tokens_buf[batch.mix_running_indices]` for the decode lane. Only `run_batch`'s CUDA branch and spec decode disaggregation write the buffer. The MLX overlap loop never does, and outside CI, the buffer is allocated with `torch.empty`, so the decode lane gathers uninitialized memory into a `torch.cat` and MPS segfaults.

Two runs with both flags crashed, at 6.07 GB and 8.90 GB free. Two runs of the identical workload without `--enable-mixed-chunk` completed cleanly. The combination is untested on this backend.

The same divergence left three per step fields unwritten on the MLX path. #32447 restored `launch_ts` and `forward_iter`; `future_map.output_tokens_buf` is the third, and this PR closes the flag combination that reads it.

## Modifications

Force disable `enable_mixed_chunk` under MLX in `_handle_mps_backends`, mirroring the existing dual chunk flash attention and diffusion LLM guards in `server_args.py`: a condition, a warning naming the backend and the reason, and the assignment.

```python
def _handle_mps_backends(self):
    if self.device == "mps":
        if not use_mlx():
            self.disable_overlap_schedule = True
        elif self.enable_mixed_chunk:
            logger.warning(
                "Mixed chunk is not supported by the MLX overlap scheduler and is disabled."
            )
            self.enable_mixed_chunk = False
```

The guard fires on `enable_mixed_chunk` alone. `is_mixed_chunk` (`scheduler.py:1066`) requires both the flag and a set `chunked_prefill_size`, so disabling the flag keeps the crashing path unreachable. Tracking one input is also simpler to reason about than two.

This is a guard, not a fix. Making the mixed path work would require the MLX overlap loop to publish decode tokens into `future_map.output_tokens_buf`, a relay it does not otherwise use, on a path with no known user. Disabling the flag is the smallest change that removes the crash.

## Accuracy Tests

This PR changes argument post processing, not model outputs.

**Unit:** `TestMlxMixedChunkGuard` in `test/registered/unit/server_args/test_server_args.py` asserts `enable_mixed_chunk` comes back `False` after `_handle_mps_backends()` under MLX and stays `True` on mps without MLX and on non-mps devices, so the guard does not overreach.

**Hardware:** `mlx-community/Qwen3-0.6B-4bit` on Apple Silicon, launched with `--enable-mixed-chunk --chunked-prefill-size 512` under `SGLANG_USE_MLX=1`. The warning logs at startup and the workload that previously crashed the scheduler now completes: a long generation in flight, an oversized prompt arriving mid decode, both requests served. Re-verified on current main after #32447 merged: the crash still reproduces unguarded, at `scheduler_mixin.py:165`.

The MLX unit suite passes at the base commit's rate. `test_metal_profiler.py` fails on the base commit as well.

## Speed Tests and Profiling

Not applicable.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations). N/A. This adds no flag and changes no documented behavior; the existing `--enable-mixed-chunk` flag is simply unsupported on this backend.
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the [SGLang code style guidance](https://docs.sglang.ai/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.ai/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

## AI Usage

The investigation and hardware verification are mine. An agent assisted with tracing the crash path and drafting this PR description. I verified every claim in this description myself.

---
### CI States

Latest PR Test (Base): :x: [Run #30417032279](https://github.com/sgl-project/sglang/actions/runs/30417032279)
Latest PR Test (Extra): :x: [Run #30417032162](https://github.com/sgl-project/sglang/actions/runs/30417032162)

### CODE REVIEW

**Comment by jlee5814** [comment]
Rebased onto current main and re-verified: the segfault still reproduces after #32447.

@yeahdongcn, could you please add the `apple-silicon` and `run-ci` labels? The gates fail at the label check, and `/tag-and-rerun-ci` did not apply them for me.

**Review by xiaolin2004 — APPROVED** [review]
LGTM. The guard is narrowly scoped, and the tests cover the MLX/MPS, non-MLX MPS, and non-MPS cases well.

Minor non-blocking nit: the test class docstring is a little more detailed than necessary, but it is also useful context and is fine as-is.

**Comment by jlee5814** [comment]
Closing ahead of the MPS redesign [#7](https://github.com/yeahdongcn/sglang/pull/7): MPS runs through `run_batch`, which writes `output_tokens_buf` on every branch, so the uninitialized read this guard rejects cannot occur. Until then, the segfault remains and this branch applies cleanly.
